src/stan/error_handling/matrix/check_symmetric.hpp

```diff
@@ -16,14 +16,20 @@ namespace stan {
     /**
      * Return <code>true</code> if the specified matrix is symmetric
      * 
+     * The error message is either 0 or 1 indexed, specified by
+     * <code>stan::error_index::value</code>.
+     *
      * NOTE: squareness is not checked by this function
```

Comment:
How could a matrix be symmetric without being square?  Shouldn't we also test squareness here?